xray/src/generation.rs

```diff
@@ -471,21 +472,34 @@ pub struct Tile {
 
 pub fn xray_from_points(
     point_cloud_client: &PointCloudClient,
-    global_from_query: &Option<Isometry3<f64>>,
+    query_from_global: &Option<Isometry3<f64>>,
     bbox: &Aabb3<f64>,
     png_file: &Path,
     image_size: Vector2<u32>,
     mut coloring_strategy: Box<dyn ColoringStrategy>,
     tile_background_color: Color<u8>,
 ) -> bool {
     let mut seen_any_points = false;
-    let point_location = PointQuery {
+    let location = match query_from_global {
+        Some(query_from_global) => {
+            let global_from_query = query_from_global.inverse();
+            PointLocation::Obb(Obb::from(bbox).transformed(&global_from_query))
+        }
+        None => PointLocation::Aabb(*bbox),
+    };
+    let point_query = PointQuery {
         attributes: coloring_strategy.attributes(),
-        location: PointLocation::Aabb(*bbox),
-        global_from_query: global_from_query.clone(),
+        location,
     };
-    let _ = point_cloud_client.for_each_point_data(&point_location, |points_batch| {
+    let _ = point_cloud_client.for_each_point_data(&point_query, |mut points_batch| {
         seen_any_points = true;
+        if let Some(query_from_global) = query_from_global {
+            points_batch.position = points_batch
+                .position
+                .iter()
+                .map(|p| (query_from_global * &Point3::from_vec(*p)).to_vec())
```

Comment:
Is this performance-critical enough that we might want to use `iter_mut()` and `std::mem::replace()`?

Reply:
It's been like this before, but I changed it to a mutable loop...